client/src/modules/Schedule/components/TableView/TableView.test.tsx

```diff
@@ -155,13 +154,34 @@ describe('TableView', () => {
         hiddenStatus: CourseScheduleItemDtoStatusEnum;
       }) => {
         const data = generateCourseData(2, [selectedStatus, hiddenStatus]);
-        render(<TableView {...PROPS_MOCK} data={data} statusFilter={selectedStatus} />);
+        render(<TableView settings={PROPS_SETTINGS_MOCK} data={data} statusFilter={selectedStatus} />);
 
         const filteredItem = screen.queryByText(new RegExp(hiddenStatus, 'i'));
         expect(filteredItem).not.toBeInTheDocument();
       },
     );
   });
+
+  it.each`
+    tag
+    ${TagsEnum.Coding}
+    ${TagsEnum.Test}
+    ${TagsEnum.Interview}
+  `('should check filters in dropdown when tag "$tag" was selected', async ({ tag }: { tag: string }) => {
+    jest
+      .spyOn(ReactUse, 'useLocalStorage')
+      // Mock useLocalStorage for tagFilter
+      .mockReturnValueOnce([[TagsEnum.Coding, TagsEnum.Test, TagsEnum.Interview], jest.fn, jest.fn]);
+    render(<TableView settings={PROPS_SETTINGS_MOCK} data={generateCourseData()} />);
+
+    const tagFilterBtn = screen.getByRole('button', { name: /filter/i });
+    fireEvent.click(tagFilterBtn);
+
+    const filtersDropdown = await screen.findByRole('menu');
+    const menuItem = within(filtersDropdown).getByRole('menuitem', { name: new RegExp(tag, 'i') });
+    const checkbox = within(menuItem).getByRole('checkbox');
+    expect(checkbox).toBeChecked();
+  });
 });
 
 function generateCourseData(
```

Comment:
This function looks convenient, but in general it would be worth avoiding complex logic in test helper functions if we do not want to write tests for tests :)

Reply:
In my opinion, this function is not so complicated and allows to reach code readability in test itself, but I got your point